src/memory/AmdFlash.cc

```diff
@@ -398,13 +399,25 @@ void AmdFlash::AmdCmd::serialize(Archive& ar, unsigned /*version*/)
 	             "value",   value);
 }
 
+// version 1: Initial version.
+// version 2: Added vppWpPinLow.
+// version 3: Changed cmd to static_vector, added status.
 template<typename Archive>
 void AmdFlash::serialize(Archive& ar, unsigned version)
 {
-	ar.serialize("ram",    *ram,
-	             "cmd",    cmd,
-	             "cmdIdx", cmdIdx,
-	             "state",  state);
+	ar.serialize("ram", *ram);
+	if (ar.versionAtLeast(version, 3)) {
+		uint8_t status = 0x80;
```

Comment:
> I assume that status is a mockup for the corresponding field in your other PR? (And that you won't push it like this?)

It’s a placeholder, since I wanted to only up the version number once. I do want to push it like this, but in a commit in the other PR I will replace this local variable with a class member variable. Like:

```diff
diff --git a/src/memory/AmdFlash.cc b/src/memory/AmdFlash.cc
index 164116976..5bc69b2cc 100644
--- a/src/memory/AmdFlash.cc
+++ b/src/memory/AmdFlash.cc
@@ -407,7 +407,6 @@ void AmdFlash::serialize(Archive& ar, unsigned version)
 {
        ar.serialize("ram", *ram);
        if (ar.versionAtLeast(version, 3)) {
-               uint8_t status = 0x80;
                ar.serialize("cmd",    cmd,
                             "status", status);
        } else {
diff --git a/src/memory/AmdFlash.hh b/src/memory/AmdFlash.hh
index 0d9a40990..745faa905 100644
--- a/src/memory/AmdFlash.hh
+++ b/src/memory/AmdFlash.hh
@@ -116,6 +116,7 @@ private:
        static constexpr unsigned MAX_CMD_SIZE = 8;
        static_vector<AmdCmd, MAX_CMD_SIZE> cmd;
        State state = State::IDLE;
+       uint8_t status = 0x80;
        bool vppWpPinLow = false; // true = protection on
 };
 SERIALIZE_CLASS_VERSION(AmdFlash, 3);
```

It has no meaning before that commit so currently it’s just discarded when deserialized. It just is there to make sure that until then, when an AmdFlash object with version 3 is serialised, the serialised data has a status field containing the default value.

Though I guess I might as well just add the member variable to the object right there to begin with, and skip the temp variable here, even if it’s not being used yet by the object.

Reply:
That's OK. 
It's also OK if you spread the changes over multiple commits if they are 'close' together (e.g. within the same push, or even a few days apart).